jwtclient/client.go

```diff
@@ -0,0 +1,22 @@
+package jwtclient
+
+import (
+	"errors"
+	"net/http"
+)
+
+// NewJWTClient is http.Client decorator which returns a client with `JWTTransport` transport.
+// `JWTTransport` will wrap the request, generate JWT token and updated Authorization header.
+func NewJWTClient(client *http.Client, opts ...Option) (*http.Client, error) {
```

Comment:
dislike this interface. you're only customizing the `client.Transport` so there's actually no need to accept a client as a parameter here. a more narrow interface would be: 
```
func NewRoundTripper(rt http.RoundTripper, opts ...Option)(http.RoundTripper, error)
```